clang/lib/DPCT/PatternRewriter.cpp

```diff
@@ -474,13 +569,63 @@ bool fixLineEndings(const std::string &Input, std::string &Output) {
   return isCRLF;
 }
 
+std::string convertCmakeCommandsToLower(const std::string &InputString) {
+  std::stringstream OutputStream;
+
+  const auto Lines = split(InputString, '\n');
+  std::vector<std::string> Output;
+  for (auto Line : Lines) {
+
+    int Size = Line.size();
+    int Index = 0;
+    for (; Index < Size && isWhitespace(Line[Index]); Index++) {
+    }
+    int Begin = Index;
+    for (Index = Begin + 1;
+         Index < Size && !isWhitespace(Line[Index]) && Line[Index] != '(';
+         Index++) {
+    }
+    int End = Index;
+    if (Index < Size && Line[Index] == '(') {
+      std::string Str = Line.substr(Begin, End - Begin);
+      std::transform(Str.begin(), Str.end(), Str.begin(),
+                     [](unsigned char c) { return std::tolower(c); });
+      if (cmake_commands.find(Str) != cmake_commands.end()) {
+        for (int Idx = Begin; Idx < End; Idx++) {
+          Line[Idx] = Str[Idx - Begin];
+        }
+      }
+    }
+
+    OutputStream << Line << "\n";
+  }
+
+  return OutputStream.str();
+}
+
+int skipCmakeComments(std::ostream &OutputStream, const std::string &Input,
+                      int Index) {
+  const int Size = Input.size();
+  if (Input[Index] == '#') {
+    for (; Index < Size && Input[Index] != '\n'; Index++) {
```

Comment:
what will happen on Windows style \r\n ?

Reply:
Input has been processed to conver "\r\n" to "\n" before this function call, so only need to check "\n"